pandas/core/internals.py

```diff
@@ -995,7 +1001,7 @@ def coerce_to_target_dtype(self, other):
 
         if self.is_bool or is_object_dtype(dtype) or is_bool_dtype(dtype):
             # we don't upcast to bool
-            return self.astype(object)
+            return self.astype(object, copy=copy)
```

Comment:
I guess this is ok, though numpy ignores the ``copy=`` flag when dtype is ``object`` so no point in passing it if dtype is object